core/src/main/java/com/linecorp/armeria/client/ResponseAs.java

```diff
@@ -182,4 +222,8 @@ public boolean requiresAggregation() {
             }
         };
     }
+
+    default <V> ConditionalResponseAs<T, R, V> andThen(ResponseAs<R, V> responseAs, Predicate<R> predicate) {
```

Comment:
Taking a second look, I think it's better if we don't expose the concrete class `ConditionalResponseAs` and expose an interface instead.

What do you think of:
1. Renaming `ConditionalResponseAs` to `DefaultConditionalResponseAs`
2. Modifying `DefaultConditionalResponseAs` to be package private
3. Create an interface `ConditionalResponseAs` and define `andThen`, `orElse` methods
4. Modify `DefaultConditionalResponseAs` to implement `ConditionalResponseAs`

Reply:
Just curious.
In what situations do we consider renaming class to `Default...` and make an interface?